pywatts/core/step.py

```diff
@@ -122,24 +133,30 @@ def _compute(self, start, end):
         input_data = self._get_input(start, end)
         target = self._get_target(start, end)
         if self.current_run_setting.computation_mode in [ComputationMode.Default, ComputationMode.FitTransform,
-                                                         ComputationMode.Train] and (
-                not self.train_if or self.train_if(input_data, target)):
+                                                         ComputationMode.Train]:
             # Fetch input_data and target data
-            start_time = time.time()
             if self.batch_size:
                 input_batch = self._get_input(end - self.batch_size, end)
                 target_batch = self._get_target(end - self.batch_size, end)
+                start_time = time.time()
                 self._fit(input_batch, target_batch)
+                self.training_time.set_kv("", time.time() - start_time)
             else:
+                start_time = time.time()
                 self._fit(input_data, target)
-            self.training_time.set_kv("", time.time() - start_time)
+                self.training_time.set_kv("", time.time() - start_time)
         elif self.module is BaseEstimator:
             logger.info("%s not fitted in Step %s", self.module.name, self.name)
 
         start_time = time.time()
         result = self._transform(input_data)
         self.transform_time.set_kv("", time.time() - start_time)
-        return result
+        if result is None:
+            return result
+        index = list(result.values())[0].indexes[_get_time_indexes(result)[0]]
+        start = max(index[0], start.to_numpy())
+        return {key: b.sel(**{_get_time_indexes(result)[0]: index[(index >= start)]}) for
+                key, b in result.items()}
```

Comment:
I don't understand why lines 154-159 are required. If running the example, `result` and the return in 159 are always the same. Please add a comment to explain the difference.

Reply:
This is a mistake, it should only return values with indexes greater than start. In online simulation, this is necessary since sometimes modules need also data before the start timestamp. This values should be removed in the result of the module/step.